### PR TITLE
WIP: Added shield pref for shieldStudy.expireSeconds to easy expiration debugging

### PR DESCRIPTION
**Pull request by gregglind**

### CODE REVIEW

File: docs/expiration.md [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+- per Engineering Notes earlier, it is up to the study to react to the ending, cleanup, and uninstall.
+
+
+### About `delayInMinutes` (calculation and semantics)
+
+General QA and verification
+
+- remember that these are alarms are usually set *once* at addon startup.  They should not react to changes mid-session.
+- `shieldStudies.logLevel = all`, will print a lot about "expires", including what the answers and logic *would be*.
+- changing system clocks should work realistically.  If you do that, you wi
```
Comment (pdehaan) [inline docs/expiration.md]: I think we're missing some words here.

File: docs/expiration.md [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+
+QA can (current) simulate this by:
+1.  install the addon once.
+2.  change the pref (mid-session), after `setup()` has been called.
+3.  restart firefox. 
+
+
+### Why is QA challenging?
+
+- Studies are expected to setup *exectly once* per session.
```
Comment (pdehaan) [inline docs/expiration.md]: typo: exactly

File: test/functional/browser.study.api.js [hunk 3]
```diff
@@ -561,6 +561,7 @@ describe("PUBLIC API `browser.study` (not specific to any add-on background logi
       }, thisSetup);
       // console.debug(full(data));
       const { info } = data;
+      console.log(info);
```
Comment (motin) [inline test/functional/browser.study.api.js]: Needs to be removed

File: webExtensionApis/study/src/studyUtils.js [hunk 4]
```diff
@@ -489,8 +524,8 @@ class StudyUtils {
     this.throwIfNotSetup("firstSeen uses telemetry.");
     logger.debug(`attempting firstSeen`);
     this._internals.isFirstRun = true;
-    await this._telemetry({ study_state: "enter" }, "shield-study");
     this.setFirstRunTimestamp(Date.now());
+    await this._telemetry({ study_state: "enter" }, "shield-study");
```
Comment (motin) [inline webExtensionApis/study/src/studyUtils.js]: Is this change related to the new pref, or is it a bugfix for something else?

File: README.md [hunk 5]
```diff
@@ -290,12 +290,13 @@ Goal: Use `webdriver` to exercise the `browser.study` API to prove correctness.
 
   (See telemetry above)
 
-  * IsFirstRun => user doesn't have the pref set for `{id}.firstRunTimestamp`
+  * IsFirstRun => user doesn't have the pref set for `shield.{id}.firstRunTimestamp`
 
   * IIF first run, then allowEnroll can make the user `ineligible`
```
Comment (motin) [inline README.md]: Nit: IF

File: webExtensionApis/study/schema.json [hunk 6]
```diff
@@ -23,7 +23,10 @@
             "type": "string"
           }
         ],
-        "testcases": [null, "a string"]
+        "testcases": [
+          null,
+          "a string"
+        ]
```
Comment (motin) [inline webExtensionApis/study/schema.json]: Is `npm run format` run on this committed code?

**Comment by motin** [comment]
Closing since the work is not in progress

**Pull request closed without merging**